src/main/steps/tasklistresponse/legalrepresentation/start/LegalRepresentationPostController.ts

```diff
@@ -33,25 +33,32 @@ export default class LegalRepresentationPostController extends PostController<An
       });
       super.redirect(req, res, redirectUrl);
     } else if (formData.legalRepresentation) {
-      req.session.userCase.legalRepresentation = formData.legalRepresentation;
-      let returnUrl = LEGAL_REPRESENTATION_SOLICITOR_NOT_DIRECT;
-      if (formData.legalRepresentation === YesOrNo.YES) {
-        returnUrl = LEGAL_REPRESENTATION_SOLICITOR_DIRECT;
-      }
-      super.redirect(req, res, returnUrl);
-      req.session.userCase?.respondents?.forEach((respondent: Respondent) => {
-        if (respondent?.value?.user?.idamId === req.session?.user.id) {
-          respondent.value.response.legalRepresentation = formData.legalRepresentation;
+      const { user, userCase } = req.session;
+      const partyType = getCasePartyType(userCase, user.id);
+      const partyDetails = getPartyDetails(userCase, user.id);
+      const client = new CosApiClient(user.accessToken, 'https://return-url');
+
+      if (partyDetails) {
+        Object.assign(partyDetails.response, { legalRepresentation: formData.legalRepresentation });
+        try {
+          req.session.userCase = await client.updateCaseData(
+            user,
+            userCase.id,
+            partyDetails,
+            partyType,
+            userCase.caseTypeOfApplication as CaseType,
+            CaseEvent.LEGAL_REPRESENTATION
+          );
+          mapDataInSession(req.session.userCase, user.id);
+          let returnUrl = LEGAL_REPRESENTATION_SOLICITOR_NOT_DIRECT;
+          if (formData.legalRepresentation === YesOrNo.YES) {
```

Comment:
check if this condition is worth checking at the start before assigning legalRepresentation to partyDetails response, if LEGAL_REPRESENTATION_SOLICITOR_DIRECT is redirecting somewhere outside this application.